MyTriggers/framework/classes/MyTriggers.cls

```diff
@@ -26,6 +26,12 @@ global virtual class MyTriggers {
     @TestVisible
     private static String triggerEventMock;
 
+    /**
+     * Used to store type of current instance for the recursion control in the variable alreadyUpdatedIdsPerClass
+     * this is required due to some limitations in apex, which don't allow retrieving the type of an instance
```

Comment:
I disagree. For the purposes of your implementation, I think the following suffices and would result in cleaner / shorter implementation. 
```
aClass superDuperClassInstance = new aClass();
String className = String.valueOf(superDuperClassInstance).substringBefore(':');
system.assertEquals(className,String.valueOf(aClass.class),'Name matches');
```

Reply:
True, would work as long as no one implements aClass like the following way. Depends on your preference - I can change it.

public class aClass  {
    public override String toString()
	{
		return 'he:ll:o world';
	}
}

Reply:
Please change.